core/src/utils.ts

```diff
@@ -123,3 +125,60 @@ export function buildUrl(host: string, path: string) {
   const url = [host, path].join('/')
   return url.replace(/([^:]\/)\/+/g, '$1')
 }
+
+// axios errors defined in official repo (https://github.com/axios/axios#error-types)
+const handleAxiosError = (e) => {
+  if (e.code == 'ERR_BAD_OPTION_VALUE') {
+    throw new OraklError(OraklErrorCode.AxiosBadOptionValue)
+  } else if (e.code == 'ERR_BAD_OPTION') {
+    throw new OraklError(OraklErrorCode.AxiosBadOption)
+  } else if (e.code == 'ECONNABORTED' || e.code == 'ETIMEDOUT') {
+    throw new OraklError(OraklErrorCode.AxiosTimeOut)
+  } else if (e.code == 'ERR_NETWORK') {
+    throw new OraklError(OraklErrorCode.AxiosNetworkError)
+  } else if (e.code == 'ERR_FR_TOO_MANY_REDIRECTS') {
+    throw new OraklError(OraklErrorCode.AxiosTooManyRedirects)
+  } else if (e.code == 'ERR_DEPRECATED') {
+    throw new OraklError(OraklErrorCode.AxiosDeprecated)
+  } else if (e.code == 'ERR_BAD_RESPONSE') {
+    throw new OraklError(OraklErrorCode.AxiosBadResponse)
+  } else if (e.code == 'ERR_BAD_REQUEST') {
+    throw new OraklError(OraklErrorCode.AxiosBadRequest)
+  } else if (e.code == 'ERR_CANCELED') {
+    throw new OraklError(OraklErrorCode.AxiosCanceledByUser)
+  } else if (e.code == 'ERR_NOT_SUPPORT') {
+    throw new OraklError(OraklErrorCode.AxiosNotSupported)
+  } else if (e.code == 'ERR_INVALID_URL') {
+    throw new OraklError(OraklErrorCode.AxiosInvalidUrl)
+  } else {
+    throw e
+  }
+}
+
+export const axiosTimeout = {
```

Comment:
It would be nice if this function can be used for any axios calls, but just the ones, where we want to define timeout. Timeout should be optional same as it is in axios. What do you think?

Reply:
Do you mean we should add timeout parameter for all url requests? such as `axiosTimeout.get(url, data, timeout)`?